logging/add-git-id/StackdriverLogWritter.cs

```diff
@@ -0,0 +1,102 @@
+ * the License.
+ */
+
+using Google.Api;
+using Google.Cloud.Logging.V2;
+using Google.Cloud.Logging.Type;
+using Google.Cloud.DevTools.Source.V1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Runtime.CompilerServices;
+using System.Reflection;
+using System.Security;
+using Google.Protobuf;
+
+namespace GoogleCloudSamples
+{
+    public static class StackdriverLogWritter
+    {
+        private const string LogId = "YOUR-LOG-ID";
+        private const string ProjectId = "YOUR-PROJECT-ID";
+
+        [MethodImpl(methodImplOptions: MethodImplOptions.NoInlining)]
+        public static string WriteLog(
+            string msg,
+            [CallerFilePath] string sourceFilePath = "",
+            [CallerLineNumber] int sourceLineNumber = 0,
+            [CallerMemberName] string memberName = "")
+        {
+            var client = LoggingServiceV2Client.Create();
+
+            var resource = new MonitoredResource { Type = "global" };
+            resource.Labels["project_id"] = ProjectId;
+
+            LogName logName = new LogName(ProjectId, LogId);
+            LogEntrySourceLocation sourceLocation = new LogEntrySourceLocation
+            {
+                File = sourceFilePath,
+                Line = sourceLineNumber,
+                Function = memberName
+            };
+
+            LogEntry logEntry = new LogEntry
+            {
+                LogName = logName.ToString(),
+                Severity = LogSeverity.Info,
+                SourceLocation = sourceLocation,
+                TextPayload = msg
+            };
+
+            IDictionary<string, string> entryLabels = new Dictionary<string, string>
+            {
+                { "size", "large" },
+                { "color", "red" }
+            };
+
+            TryAddGitRevisionId(entryLabels);
+
+            client.WriteLogEntries(
+                logName: LogNameOneof.From(logName),
+                resource: resource, 
+                labels: entryLabels, 
+                entries: new[] { logEntry }, 
+                callSettings: null);
+            return logEntry.ToString();
+        }
+
+        private static void TryAddGitRevisionId(IDictionary<string, string> labels)
+        {
+            try
+            {
+                var gitId = SourceContext.AppSourceContext?.Git?.RevisionId;
+                if (!String.IsNullOrWhiteSpace(gitId))
+                {
+                    labels.Add(SourceContext.GitRevisionIdLogLabel, gitId);
+                }
+            }
+            catch (Exception ex) when (
+                ex is SecurityException
+                || ex is InvalidProtocolBufferException
+                || ex is InvalidJsonException
+                || ex is UnauthorizedAccessException)
+            {
+                // This is best-effort only, exceptions from reading/parsing the source_context.json are ignored.
```

Comment:
If user incorporate the sample into their production code, I don't want the logger blow up their products.   I added Debug.Writeline(..)  , hopefully it helps in some extent.